src/main/java/com/fruits/ecommerce/controller/CartController.java

```diff
@@ -26,9 +27,10 @@ public ResponseEntity<Void> addProductToCart(
         return ResponseEntity.status(HttpStatus.CREATED).build();
     }
     @GetMapping
-    @PreAuthorize("hasAnyRole('CUSTOMER','USER')")
+    @PostAuthorize("hasRole('CUSTOMER')")
```

Comment:
This role can be added to Security configuration instead for the path "**/cart/**"

Reply:
you are right i will change it